net2cog/utilities.py

```diff
@@ -384,3 +387,144 @@ def is_valid_spatial_dimensions_with_standard_name_units(
+    based on the ValueError message.  Raises a ValueError if
+    no matching handler is found.
+
+    Parameters
+    ----------
+    nc_xarray : xarray.DataTree
+        DataTree object representing the root group of the NetCDF-4 file.
+    variable_path: str
+        Variable path is present in DataTree
+    value_error_message: str
+        The ValueError exception message
+
+    Returns
+    -------
+        Callable: Returns the right callable method
+        apply_fillvalue_to_missing_value()
+        or any other process.
+
+    """
+    # ValueError: Variable None has conflicting _FillValue (255) and
+    # missing_value (200).  Cannot encode data.
+    fill_missing_value_keywords = ['_FillValue', 'missing_value']
+
+    if all(word in value_error_message for word in fill_missing_value_keywords):
+        fill_value, missing_value = (
+            get_fillvalue_and_missing_value(nc_xarray, variable_path)
+        )
+
+        if fill_value is not None and missing_value is not None:
+            return apply_fillvalue_to_missing_value
+
+    raise ValueError(value_error_message)
+
+
+def apply_fillvalue_to_missing_value(
+        nc_xarray: xr.DataTree, variable_path: str
+) -> xr.DataTree:
+    """This function replaces occurrences of missing_value in the variable's
+    data array with _FillValue. It also removes the missing_value attribute
+    and adds a new process_note attribute to document the transformation
+    for reference.
+
+    Parameters
+    ----------
+    nc_xarray : xarray.DataTree
+        DataTree object representing the root group of the NetCDF-4 file.
+    variable_path: str
+        Variable path is present in DataTree
+
+    Returns
+    -------
+    xr.DataTree
+        New DataTree with missing_value data replace with _FillValue data,
+        missing_value attribute delete, and new process_note attribute
+        to explain the process.
+
+    """
+    fill_value, missing_value = (
+        get_fillvalue_and_missing_value(nc_xarray, variable_path)
+    )
+
+    if fill_value is None or missing_value is None:
+        raise ValueError("Missing _FillValue or missing_value attribute.")
+
+    # DataTree nc_xarray is immutable so copy new DataTree
+    # to change missing_value data
+    nc_xarray_tmp = nc_xarray.copy()
+    values_tmp = nc_xarray_tmp[variable_path].values.copy()
+
+    # Use np.nditer with write access
+    with np.nditer(values_tmp, flags=['multi_index'], op_flags=['readwrite']) as values_iter:
+        for values in values_iter:
+            if values[...] == missing_value:
+                values[...] = fill_value
+
+    # Assign updated values back to the DataTree
+    nc_xarray_tmp[variable_path].values = values_tmp
+
+    # Delete missing_value
+    if 'missing_value' in nc_xarray_tmp[variable_path].encoding:
+        del nc_xarray_tmp[variable_path].encoding['missing_value']
+    elif 'missing_value' in nc_xarray_tmp[variable_path].attrs:
```

Comment:
Is it mutually exclusive that `missing_value` would only be in either the encoding or the attributes. Could it ever be in both for some other non-MODIS collection? (Feels like you don't lose anything by making this `elif` an `if`, and covering that just in case.)